tests/scalers/azure/azure_aad_pod_identity/azure_aad_pod_identityte_test.go

```diff
@@ -0,0 +1,152 @@
+	triggerAuthNilIDName          = fmt.Sprintf("%s-ta-nil", testName)
+	clusterTriggerAuthEmptyIDName = fmt.Sprintf("%s-cta-empty", testName)
+	clusterTriggerAuthNilIDName   = fmt.Sprintf("%s-cta-nil", testName)
+)
+
+type templateData struct {
+	TestNamespace                 string
+	TriggerAuthEmptyIDName        string
+	TriggerAuthNilIDName          string
+	ClusterTriggerAuthEmptyIDName string
+	ClusterTriggerAuthNilIDName   string
+}
+
+const (
+	triggerAuthEmptyIDTemplate = `
+apiVersion: keda.sh/v1alpha1
+kind: TriggerAuthentication
+metadata:
+  name: {{.TriggerAuthEmptyIDName}}
+  namespace: {{.TestNamespace}}
+spec:
+  podIdentity:
+    provider: azure
+    identityId: ""
+`
+
+	triggerAuthNilIDTemplate = `
+apiVersion: keda.sh/v1alpha1
+kind: TriggerAuthentication
+metadata:
+  name: {{.TriggerAuthNilIDName}}
+  namespace: {{.TestNamespace}}
+spec:
+  podIdentity:
+    provider: azure
+`
+	clusterTriggerAuthEmptyIDTemplate = `
+apiVersion: keda.sh/v1alpha1
+kind: ClusterTriggerAuthentication
+metadata:
+  name: {{.ClusterTriggerAuthEmptyIDName}}
+spec:
+  podIdentity:
+    provider: azure
+    identityId: ""
+`
+
+	clusterTriggerAuthNilIDTemplate = `
+apiVersion: keda.sh/v1alpha1
+kind: ClusterTriggerAuthentication
+metadata:
+  name: {{.ClusterTriggerAuthNilIDName}}
+spec:
+  podIdentity:
+    provider: azure
+`
+)
+
+func TestScaler(t *testing.T) {
+	// setup
+	t.Log("--- setting up ---")
+
+	// Create kubernetes resources
+	kc := GetKubernetesClient(t)
+	data, templates := getTemplateData()
+
+	CreateKubernetesResources(t, kc, testNamespace, data, templates)
+
+	// test auth
+	testTriggerAuthenticationWithEmptyID(t, data)
+	testTriggerAuthenticationWithNilID(t, data)
+	testClusterTriggerAuthenticationWithEmptyID(t, data)
+	testClusterTriggerAuthenticationWithNilID(t, data)
+
+	// cleanup
+	DeleteKubernetesResources(t, testNamespace, data, templates)
+}
+
+func getTemplateData() (templateData, []Template) {
+	return templateData{
+		TestNamespace:                 testNamespace,
+		TriggerAuthEmptyIDName:        triggerAuthEmptyIDName,
+		TriggerAuthNilIDName:          triggerAuthNilIDName,
+		ClusterTriggerAuthEmptyIDName: clusterTriggerAuthEmptyIDName,
+		ClusterTriggerAuthNilIDName:   clusterTriggerAuthNilIDName,
+	}, []Template{}
+}
+
+// expect triggerauthentication should not be created with empty identity id
+func testTriggerAuthenticationWithEmptyID(t *testing.T, data templateData) {
+	t.Log("--- create triggerauthentication with empty identity id  ---")
+
+	err := KubectlApplyWithErrors(t, data, "triggerAuthEmptyIDTemplate", triggerAuthEmptyIDTemplate)
+	assert.Errorf(t, err, "can deploy TriggerAuthtication - %s", err)
+}
+
+// expect triggerauthentication can be created without identity id property
+func testTriggerAuthenticationWithNilID(t *testing.T, data templateData) {
+	t.Log("--- create triggerauthentication with nil identity id  ---")
+
+	kedaKc := GetKedaKubernetesClient(t)
```

Comment:
In other tests, we pass the client as parameter from the main execution, even though isn't mandatory, I'd do the same for consistency

Reply:
I kept the client as a parameter before. But it cannot pass the static check. Is there any way to skip that?

Reply:
What do you mean? We are already doing it in other places, for example:
https://github.com/kedacore/keda/blob/b4919ea956b2ce723c35012b084045690f7212f5/tests/scalers/azure/azure_data_explorer/azure_data_explorer_test.go#L147-L158

Reply:
When I pass the client as parameter to keep consistnecy, I cannot pass the check as the parameter is unused.

<img width="1119" alt="image" src="https://github.com/kedacore/keda/assets/4278354/c2a93b5c-d72c-4ebb-929d-e3e7efb7a727">